test/economic-resource/resource_links.js

```diff
@@ -113,8 +113,8 @@ test('EconomicResource composition / containment functionality', async (t) => {
     readResource = readResp.economicResource
     t.deepLooseEqual(readResource.id, resourceId1, 'container resource re-retrieval OK')
     t.equal(readResource.contains && readResource.contains.length, 2, 'container resource reference appended')
-    t.deepLooseEqual(readResource.contains && readResource.contains[0], resourceId2, 'container resource reference B OK')
```

Comment:
The ordering of this seems to be the cause of the failing test :s

Reply:
Pretty sure this is due to the following relationship, where reciprocals are declared in the same zome: https://github.com/h-REA/hREA/blob/sprout/zomes/rea_economic_resource/zome_idx_observation/src/lib.rs#L16-L17

I do seem to be observing similar issues with `EconomicEvent.affects` / `EconomicResource.affectedBy`, but they aren't showing up in tests (purely coincidentally). This also means we're still relying on duplicate indexes on the DHT in order to correctly query results.

I see two paths forward:

1. Delay this PR for now and revisit the scope to include index link validations outlined as part 1 of [this comment](https://github.com/h-REA/hREA/issues/264#issuecomment-1224314138).
2. Merge this as-is and log a bug for undefined ordering of locally-adjacent record indexes (eg. EconomicEvent <-> EconomicResource, Process <-> EconomicEvent, Proposal <-> ProposedIntent). Address the new bug with validation rules as per 1) above.

Either course of action would require migrations to the index integrity zomes format, but 2) results in more immediate gains of modest WASM bytesize savings.

Reply:
Hey sorry, I've been thinking about this. 
I think option `1` is better for this MMR release. The issue sounds interlinked with other ones and so a comprehensive sweep over them in a different work pass/scope sounds useful to get the job done proper.

Reply:
Is the fix in this commit useful to merge, on its own?
https://github.com/h-REA/hREA/pull/363/commits/f55783416fc0da9b108e5ca2c1599c0a541e069e

Reply:
nah, doesn't matter. With the integrity zomes separated there can be no conflicts anyway, even though there will technically be conflicting/redundant links in different zomes within the same DHT.